internal/genconfig/generate_config.go

```diff
@@ -592,7 +593,7 @@ func ctyCollectionValues(val cty.Value) []cty.Value {
 func hclEscapeString(str string) string {
 	// TODO: Replace this with more complete HCL logic instead of the simple
```

Comment:
Isn't this done now?

Reply:
see Martin's comment here for a more thorough explanation of what I'd consider to be done: https://github.com/hashicorp/terraform/pull/35754#issuecomment-2372449103

Basically, it's not the detection that this TODO is talking about but the escaping itself - it's not always enough to just wrap it in quotes.

Reply:
Ah ok 👍